packages/markers/src/browser/marker-manager.ts

```diff
@@ -183,9 +182,13 @@ export abstract class MarkerManager<D extends object> {
 
     protected internalSetMarkers(uri: URI, owner: string, data: D[]): Marker<D>[] {
         const collection = this.getCollection(uri);
-        const result = collection.setMarkers(owner, data);
+        const old = collection.setMarkers(owner, data);
+        const result = collection.getMarkers(owner);
```

Comment:
`result` should be named `current` or something similar.

Reply:
Could we get rid of this line and instead use ```if (data.length === 0)``` to find out if the marker list is empty? This would avoid the extra call to get the updated markers list.